chapter0-What-is-Java/terminology.md

```diff
@@ -1,19 +1,21 @@
+**Object File:** The result from compiling source code. Object files can have different formats and may also depend on other object files or programs.  Hence, some will not be directly executable without further compilation and / or in isolation.  Several object files, which depend on each other to create a fully-functioning program, can be linked together to create an executable file.
 
-**Interpreter** An interpreted language takes each high-level statement, determines its low-level version and executes (while linking if need be) the result. This is done for each statement in succession (before the next high-level statement is even looked at). This  conversion is "on-the-fly", also called "as needed".
+**Linker:** Links a collection of object files to form an executable file.
+
+**Executable file:** A file that the operating system can load into memory and run directly without further compilation.  Executable files are sometimes known as 'relocatable' since they can be moved to and run on any instance of the intended platform (see 'platform', below).
 
-**Bytecode** The result from the Java compiler (often called the JVM compiler). Essentially it is an object file for a virtual machine, which allows it to be used on any machine that has Java. (However, bytecode is not an executable file.  To execute a bytecode file, you need to invoke a Java interpreter).
+**Interpreter** An interpreted language takes each high-level statement, determines its low-level version and executes (while linking if need be) the result. This is done for each statement in succession (before the next high-level statement is even looked at). This  conversion is "on-the-fly", also called "as needed".
 
-**Machine code:** The result from linking compiled source code (a collection of object files) together
+**Source code** A high level programming language, such as Java or Ruby.
 
-**Source code** A high level programming language
+**Bytecode** The result from the Java compiler (often called the JVM compiler). Essentially, it is an object file for a virtual machine, which allows it to be used on any machine that has Java. Bytecode is not, however, an executable file.  To execute a bytecode file, you need to invoke a Java interpreter.
 
-**Platform** A specific target machine that code can be executed on. For example, if you compile a C++ program on a Windows machine, the executable file can be copied to any other machine but it will only run on other Windows machines but never another machine (e.g., a Mac or a Linux machine). A platform is determined by the target machine (along with its operating system).
+**Machine code:** The result from linking compiled source code (a collection of object files) together.  I.e. The code which comprises executable files.
```

Comment:
I don't understand how this definition is different from executable. To me, machine code is just low-level code, or code that machine can understand, but I may be missing some java terminology here.

Reply:
I suspect that I have something slightly wrong.  By 'executable' I mean a whole program and by 'machine code' I mean the code which is contained within an executable.

Reply:
I've left this one in for now, given the time pressure.  I think that I can clarify any misunderstanding/s that crop up with the students.  It would be cool to discuss in person.